.github/workflows/platform-pull-request.yml

```diff
@@ -47,58 +47,137 @@ jobs:
       - name: Cloning repo
         uses: actions/checkout@v3
 
-      - name: Run Local API
-        id: run-local-api
-        uses: ./.github/actions/run-local-api
+      - name: Run E2E tests against local
+        uses: ./.github/actions/local-e2e-tests
         with:
-          e2e_test_token: some-token
-          # As per https://stackoverflow.com/q/65497331/421808 172.17.0.1 seems like the only way to resolve host DB
-          database_url: postgres://postgres:postgres@172.17.0.1:5432/flagsmith
           sentry_sdk_dsn: ${{ secrets.SENTRY_SDK_DSN }}
           sendgrid_api_key: ${{ secrets.SENDGRID_API_KEY }}
-          disable_analytics_features: true
+          slack_token: ${{ secrets.SLACK_TOKEN }}
+          tests: segment-part-1 environment
+          concurrency: 1
+
+  run-e2e-tests-2:
+    runs-on: ubuntu-latest
+    name: E2E Segment Part 2
+
+    services:
+      postgres:
+        image: postgres:11.12-alpine
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_DB: flagsmith
+        ports: ['5432:5432']
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - name: Cloning repo
+        uses: actions/checkout@v3
 
       - name: Run E2E tests against local
-        uses: ./.github/actions/e2e-tests
+        uses: ./.github/actions/local-e2e-tests
         with:
-          e2e_test_token: some-token
+          sentry_sdk_dsn: ${{ secrets.SENTRY_SDK_DSN }}
+          sendgrid_api_key: ${{ secrets.SENDGRID_API_KEY }}
           slack_token: ${{ secrets.SLACK_TOKEN }}
-          environment: local
+          tests: segment-part-2
+          concurrency: 1
+
+  run-e2e-tests-3:
+    runs-on: ubuntu-latest
+    name: E2E segments-3, signup,flag, invite, project tests on local
```

Comment:
Its is just for efficiency. As explained in the description, segments-2 and segments-1 are the two tests that take the most, so an efficient way to arrange these tests is having 3 instances running in parallel one with the segment-2, another with segment-1 plus a small test, and a third one with all the remaining tests since they all take a bit less than segment-2 to run.